CMakeLists.txt

```diff
@@ -19,6 +19,10 @@ if(POLICY CMP0171)
   cmake_policy(SET CMP0171 NEW)
 endif()
 
+# When adjusting CMake flag variables, we must not override those explicitly
+# set by the user. These are a subset of the CACHE_VARIABLES property.
+get_directory_property(precious_variables CACHE_VARIABLES)
```

Comment:
In commit "cmake: Respect user-provided configuration-specific flags" (edde96376a2961dec3730331b3d171ddf972589f)

Would be good for the comment here to note that this needs to be called before the project() command otherwise the list will include variables not actually set by the user. (I wouldn't have known that unless I saw the discussion at https://github.com/bitcoin/bitcoin/pull/32356#discussion_r2075396465)